org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/ProjectConfigurator.java

```diff
@@ -12,8 +12,28 @@
 
 /**
  * Interface to hook into Buildship's project synchronization.
+ *
+ * <p>
+ * The primary intent of this interface is to let external plugins set up project configuration for
+ * tools that are unknown by Buildship. An implementation can be registered via the
+ * {@code projectconfigurators} extension point. An implementation is not limited to use any APIs
```

Comment:
An implementation is free to use any APIs to change the project. It is recommended that configurators only add additional configuration and not remove anything previously added (e.g. don't delete projects or dependencies)

Reply:
Thanks for the proposal, I'll adjust the javadoc before merging.